### PR TITLE
 Allow the enabling of http endpoint on the side-cars

### PR DESCRIPTION
**Pull request by outscale-mdr**
All the side-cars have the ability to expose an http endpoint in order
to accomplish two things:
- being able to access the metrics
- being able to check the health of the side-cars through
  /healthz/leader-election path

Each endpoint can be enable unitarily in order to reduce the number of
open ports.

### CODE REVIEW

File: osc-bsu-csi-driver/templates/controller.yaml [hunk 1]
```diff
@@ -154,6 +175,27 @@ spec:
...
+              protocol: TCP
+          {{- end }}
+          {{- if and .Values.sidecars.attacherImage.enableHttpEndpoint .Values.sidecars.attacherImage.enableLivenessProbe }}
+          livenessProbe:
+            failureThreshold: 5
+            httpGet:
+              path: /healthz/leader-election
+              port: httpendpoint
+              scheme: HTTP
+            initialDelaySeconds: 10
```
Comment (outscale-vbr) [inline osc-bsu-csi-driver/templates/controller.yaml:194]: How do you choose you value ?

File: osc-bsu-csi-driver/templates/controller.yaml [hunk 2]
```diff
@@ -154,6 +175,27 @@ spec:
...
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 3
```
Comment (outscale-vbr) [inline osc-bsu-csi-driver/templates/controller.yaml:197]: Why no readiness ?

**Review by jerome-jutteau — APPROVED** [review]

File: osc-bsu-csi-driver/templates/controller.yaml (hunk 1 above)
Comment (outscale-mdr) [reply osc-bsu-csi-driver/templates/controller.yaml]: Now, I use the default values. No need to configure more restrictive values

File: osc-bsu-csi-driver/templates/controller.yaml (hunk 2 above)
Comment (outscale-mdr) [reply osc-bsu-csi-driver/templates/controller.yaml]: In the sidecars documentation, it is required to configure only liveness probe. The readiness seems not necessary in our use case.

Comment (outscale-vbr) [reply osc-bsu-csi-driver/templates/controller.yaml]: https://github.com/kubernetes-csi/external-provisioner#http-endpoint
OK

File: osc-bsu-csi-driver/templates/controller.yaml (hunk 1 above)
Comment (outscale-vbr) [reply osc-bsu-csi-driver/templates/controller.yaml]: Ok

**Review by outscale-vbr — APPROVED** [review]
Ok good

**Pull request merged**